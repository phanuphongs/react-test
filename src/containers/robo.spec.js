import { shallow } from 'enzyme';
import React from 'react';
import Robo from "./robo";


let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        query: '',
        isPending: false,
    }
    wrapper = shallow(<Robo {...mockProps}/>)
})


it('expect to render Robo component', () => {
    expect(wrapper).toMatchSnapshot( )
})

it('filters robots correctly', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'john Walker',
            email: 'john@gmail.com'
        },
        {
            id: 2,
            name: 'Ashe Doe',
            email: 'ashe@gmail.com'
        }
    ]
    let query = ''
    expect(wrapper.instance().filterRobot(mockRobots, query)).toEqual(mockRobots)
    query = 'walker'
    expect(wrapper.instance().filterRobot(mockRobots, query)).toEqual([mockRobots[0]])
    query = 'erere'
    expect(wrapper.instance().filterRobot(mockRobots, query)).toEqual([])
})