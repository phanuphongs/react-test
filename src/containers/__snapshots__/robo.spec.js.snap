// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Robo component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Robo
    isPending={false}
    onRequestRobots={[Function]}
    query=""
    robots={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchBox
          searchChange={undefined}
/>,
        <Scroll>
          <ErrorBoundry>
                    <CardList
                              robots={Array []}
                    />
          </ErrorBoundry>
</Scroll>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchChange": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ErrorBoundry>
            <CardList
                        robots={Array []}
            />
</ErrorBoundry>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CardList
              robots={Array []}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "robots": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchBox
            searchChange={undefined}
/>,
          <Scroll>
            <ErrorBoundry>
                        <CardList
                                    robots={Array []}
                        />
            </ErrorBoundry>
</Scroll>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ErrorBoundry>
              <CardList
                            robots={Array []}
              />
</ErrorBoundry>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <CardList
                robots={Array []}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "robots": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
